tinyDA/chain.py

```diff
@@ -336,9 +366,13 @@ def sample(self, iterations, progressbar=True):
             else:
                 # when subsampling is complete, create a new fine link from the
                 # previous coarse link.
+                proposal_index = self._get_proposal_index()
                 proposal_link_fine = self.posterior_fine.create_link(
-                    self.chain_coarse[-1].parameters
+                    self.chain_coarse[proposal_index].parameters
                 )
+                self.promoted_coarse.append(self.chain_coarse[proposal_index]) 
+                # add effective subchain lenght to list
+                self.subchain_lengths.append(proposal_index + self.subchain_length)
```

Comment:
I think the `subchain_length` here will actually be `proposal_index + self.subchain_length + 1`, since if for example we have `proposal_index == -self.subchain_length` (the shortest possible length), the specific `subchain_length` should still be 1. I think this is a somewhat semantic point, but it should also make constructing the estimator easier down the line.